packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -214,6 +214,8 @@ export class RestApi extends RestApiRef implements cdk.IDependable {
         return new Method(this, httpMethod, { resource: this.root, httpMethod, integration, options });
       },
       addProxy: (options?: ResourceOptions) => {
+        // Must have ANY method on the root when doing this
+        this.root.addMethod('ANY');
```

Comment:
Would have been nicer if you put this logic in `ProxyResource` in case someone uses it directly. Ideally we want the `addXxx` APIs to behave exactly like using the APIGW constructs directly. You should be able to identify that "parent" is "root" by going back to the API object